dbms/programs/client/Client.cpp

```diff
@@ -1648,9 +1683,12 @@ class Client : public Poco::Util::Application
         }
 
         /// Extract settings from the options.
-#define EXTRACT_SETTING(TYPE, NAME, DEFAULT, DESCRIPTION) \
-        if (options.count(#NAME)) \
-            context.setSetting(#NAME, options[#NAME].as<std::string>());
+#define EXTRACT_SETTING(TYPE, NAME, DEFAULT, DESCRIPTION)                \
+        if (options.count(#NAME))                                        \
+        {                                                                \
+            context.setSetting(#NAME, options[#NAME].as<std::string>()); \
+            config().setString(#NAME, options[#NAME].as<std::string>()); \
```

Comment:
@abyss7 why do we put settings to `config()` here ? Looks excessive.
Maybe you actually wanted a single setting to be in `config()`, not everything?